File: insertion_sort/c/insertion_sort.c
#include <stdio.h>
#define swap(a, b) int tmp = (a); (a) = (b); (b) = tmp;

void insertion_sort(int *arr, int size)
{
  int j = 1;
  for (int i = 1; i < size; i++) {
    j = i;
    while (j > 0 && arr[j] < arr[j - 1]) {
      swap(arr[j], arr[j - 1]);
      j--;
    }
  }
}

File: insertion_sort/java/InsertionSort.java

public class InsertionSort {

  public static int[] InsertionSort(int[] array) {
  
    for (int i = 1; i < array.length; i++) {
      int j = i;
      int key = array[i];

      while (j > 0 && array[j - 1] > key) {
        array[j] = array[j - 1];
        j--;
      }

      array[j] = key;
    }

    return array;
  }
}